### PR TITLE
Add Dragon Player

### PR DESCRIPTION
**Pull request by Mailaender**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://matrix.to/#/#flatpak:kde.org
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)* https://invent.kde.org/multimedia/dragon/-/merge_requests/3

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by hfiguiere** [comment]
cc @flathub/kde

**Comment by nedrichards** [comment]
Yeah, I think not being able to open files the obvious way is a blocker for now.

**Comment by Mailaender** [comment]
I noticed that opening a file via Dolphin already works, so I submitted it.

**Comment by Mailaender** [comment]
I suggest to put it in beta.

**Comment by tsdgeos** [comment]
maybe¿ what others think @travier  ?

**Comment by Pointedstick** [comment]
Dragon has changed quite a bit since this PR was opened; can you rebase with 23.04.1 and try again, maybe? It's possible the issue with opening files from the app as gone away already.

**Comment by Mailaender** [comment]
bot, build org.kde.dragonplayer

**Comment by Mailaender** [comment]
Seems it is indeed already fixed.

**Comment by Pointedstick** [comment]
Indeed, fixed for me too.

**Comment by Pointedstick** [comment]
+1 from me

**Comment by travier** [comment]
I have

```
kf.solid.backends.udisks2: Failed enumerating UDisks2 objects: "org.freedesktop.DBus.Error.Disconnected" 
 "Not connected to D-Bus server"
kf.solid.backends.udisks2: Failed enumerating UDisks2 objects: "org.freedesktop.DBus.Error.Disconnected" 
 "Not connected to D-Bus server"
```

and unrelated to above, I wasn't able to play video files :/

**Comment by Mailaender** [comment]
Worked fine on KDE Neon using X11.

**Comment by tsdgeos** [comment]
Looks reasonably good to me

**Comment by travier** [comment]
OK, it does play some video files. Let's do it.

**Comment by travier** [comment]
CC @barthalion @nedrichards

**Comment by Mailaender** [comment]
Ping @barthalion @nedrichards Please reopen.

**Comment by Mailaender** [comment]
Ping @barthalion @nedrichards Please reopen.

File: org.kde.dragonplayer.json [hunk 1]
```diff
@@ -0,0 +1,39 @@
+{
+  "app-id": "org.kde.dragonplayer",
+  "runtime": "org.kde.Platform",
+  "runtime-version": "5.15-22.08",
```
Comment (bilelmoussaoui) [inline org.kde.dragonplayer.json:4]: should be bumped no?

File: org.kde.dragonplayer.json [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  "finish-args": [
+    "--share=ipc",
+    "--share=network",
+    "--socket=wayland",
+    "--socket=fallback-x11",
+    "--socket=pulseaudio",
+    "--device=dri",
+    "--filesystem=home",
+    "--filesystem=/run/media",
+    "--filesystem=/media",
```
Comment (bilelmoussaoui) [inline org.kde.dragonplayer.json:17]: does it not use the portal?

File: org.kde.dragonplayer.json (hunk 1 above)
Comment (travier) [reply org.kde.dragonplayer.json:4]: That's the latest KDE Runtime that we have (based on fdo 22.08)

File: org.kde.dragonplayer.json (hunk 2 above)
Comment (travier) [reply org.kde.dragonplayer.json:17]: Good point. Will give it a try

**Comment by travier** [comment]
bot, build org.kde.dragonplayer

File: org.kde.dragonplayer.json (hunk 2 above)
Comment (Erick555) [reply org.kde.dragonplayer.json:17]: Perhaps it does use file portals but portals are too limited for media apps (loading subtitles to a movie, playlists. Other published apps don't use [them](https://github.com/flathub/org.kde.elisa/blob/master/org.kde.elisa.json#L13).

There is even nasty portal [bug](https://github.com/flatpak/xdg-desktop-portal/issues/689) when it doesn't close once opened files thus prevent unmounting removable media.

Comment (Mailaender) [reply org.kde.dragonplayer.json]: Reduced the default permissions.

**Comment by Mailaender** [comment]
bot, build org.kde.dragonplayer

**Comment by travier** [comment]
Unfortunately Dragon Player does not correctly use the portals right now so this does not work thus why it needs full home access.

**Comment by barthalion** [comment]
bot, build org.kde.dragonplayer

**Comment by travier** [comment]
Hum, this crashed for me in the open dialog. I have the following in the console but I don't think that's the issue:

```
kf.kio.widgets: Cannot read information about filesystem under path "/var/home/tim/.var/app/org.kde.dragonplayer/cache/thumbnails/"
kf.solid.frontend.devicemanager: Couldn't get StorageAccess for ""/var/home/tim/.var/app/org.kde.dragonplayer/cache/thumbnails/"" - File doesn't exist
```

Looks like this still needs more fixes 😕.

**Comment by barthalion** [comment]
Seems fine on my end with the latest build.

**Comment by Erick555** [comment]
The `/media` and `/run/media` paths that were part of original manifest are missing.

**Pull request closed without merging**